components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Protocol/src/Services/SubscriptionServices.cs

```diff
@@ -1474,13 +1474,15 @@ public class MonitoredItemWrapper {
                     if (pendingAlarmsOptions.ConditionIdIndex.HasValue && pendingAlarmsOptions.RetainIndex.HasValue) {
                         var conditionId = values[pendingAlarmsOptions.ConditionIdIndex.Value].Value.ToString();
                         var retain = values[pendingAlarmsOptions.RetainIndex.Value].Value.GetValue<bool>(false);
-                        if (PendingAlarmEvents.ContainsKey(conditionId) && !retain) {
-                            PendingAlarmEvents.TryRemove(conditionId, out var monitoredItemNotificationModel);
-                            pendingAlarmsOptions.Dirty = true;
-                        }
-                        else if (retain) {
-                            pendingAlarmsOptions.Dirty = true;
-                            PendingAlarmEvents[conditionId] = monitoredItemNotification;
+                        lock (PendingAlarmEvents) {
+                            if (PendingAlarmEvents.ContainsKey(conditionId) && !retain) {
```

Comment:
I think there are issues using a dictionary for a lock and seperate lock object is recommended. Could you double check?

Reply:
created separate lock object for safety. Internet is not agreeing on this :-)